### PR TITLE
vs-eng-Public-method-to-spawn-a-game-object-LV-521

### PR DESCRIPTION
**Pull request by akalscheur-bradley**
Gives multiple ways to spawn in an object:
- Serialized fields for in-editor use
- Function overrides to call it outside of the script
Both allow for spawning with or without a parent

### CODE REVIEW

File: Assets/General/Scripts/UniversalFunctions/ObjectSpawner.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+using UnityEditor;
+using UnityEngine;
+
+/// <summary>
+/// Used as a way for designers to access prefab instantiation from the editor
+/// </summary>
+public class ObjectSpawner : MonoBehaviour
+{
+    [Tooltip("The prefab you want to spawn in")]
+    [SerializeField] GameObject _prefab;
```
Comment (RyanASwanson) [inline Assets/General/Scripts/UniversalFunctions/ObjectSpawner.cs:20]: perhaps prefab is too vague? Maybe _objectToSpawn or something like that

File: Assets/General/Scripts/UniversalFunctions/ObjectSpawner.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    [Tooltip("The position you want to spawn the object in.  " +
+             "Will be in relation to the parent if defined.  " +
+             "Otherwise will spawn in worldspace")]
+    [SerializeField] Vector3 _position;
```
Comment (RyanASwanson) [inline Assets/General/Scripts/UniversalFunctions/ObjectSpawner.cs:24]: Perhaps could be _spawnPosition instead

File: Assets/General/Scripts/UniversalFunctions/ObjectSpawner.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    [Tooltip("The rotation the object will spawn in at" +
+             "Will be in relation to the parent if defined.  " +
+             "Otherwise will spawn in worldspace")]
+    [SerializeField] Quaternion _rotation;
```
Comment (RyanASwanson) [inline Assets/General/Scripts/UniversalFunctions/ObjectSpawner.cs:28]: Perhaps could be _spawnRotation instead

File: Assets/General/Scripts/UniversalFunctions/ObjectSpawner.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    [Tooltip("(optional) Object you want the prefab instance childed to")]
+    [SerializeField] Transform _parent;
```
Comment (RyanASwanson) [inline Assets/General/Scripts/UniversalFunctions/ObjectSpawner.cs:30]: Perhaps could be _spawnedObjectParent

**Review by RyanASwanson — CHANGES_REQUESTED** [review]
Some of the variable names are a bit generic and vague. Functionality wise no complaints. I gotta get more and more picky with code as this has gone on because I have less to comment on as people get better and better.

**Review by ASLH68 — APPROVED** [review]

File: Assets/General/Scripts/UniversalFunctions/ObjectSpawner.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+/// </summary>
+public class ObjectSpawner : MonoBehaviour
+{
+    /// <summary>
+    /// Spawns an object based on the given parameters.  Used for spawning in worldspace
+    /// </summary>
+    /// <param name="prefab">The prefab you want to spawn in</param>
+    /// <param name="position">The position in worldspace you want to spawn at</param>
+    /// <param name="rotation">Rotation you want to spawn at</param>
+    public void SpawnObject(GameObject prefab, Vector3 position, Quaternion rotation)
```
Comment (RyanASwanson) [inline Assets/General/Scripts/UniversalFunctions/ObjectSpawner.cs:25]: Still think that these names can specify 'spawnPosition' and 'spawnRotation' but regardless I approve

**Review by RyanASwanson — APPROVED** [review]
Left a suggestion, not a requirement. I approve

**Review by AlecatEnjoyer — APPROVED** [review]
I approve

**Pull request merged**